File: credit_card.py
try:
        card_number = int(input("Enter the credit card number: "))
        card_number_str = str(card_number)
        if len(card_number_str) < 13 or len(card_number_str) > 16:
            print("Invalid credit card number!")
        else:
            card_type = ""
            if card_number_str.startswith("4"):
                card_type = "Visa"
            elif card_number_str.startswith("5"):
                card_type = "MasterCard"
            elif card_number_str.startswith("37"):
                card_type = "American Express"
            elif card_number_str.startswith("6"):
                card_type = "Discover"
            else:
                card_type = "unknown"

            odd_sum = 0
            even_sum = 0

            for i in range(len(card_number_str) - 1, -1, -2):
                odd_sum += int(card_number_str[i])

            for i in range(len(card_number_str) - 2, -1, -2):
                digit = 2 * int(card_number_str[i])
                even_sum += (digit // 10) + (digit % 10)

            total_sum = odd_sum + even_sum
            if total_sum % 10 == 0:
                print(f"Credit Card Type: {card_type}")
                print(f"Credit Card Number: {card_number}")
                print(f"Credit Card Length: {len(card_number_str)}")
                print("Credit Card Validity Status: valid")
            else:
                print("Credit Card Type: Invalid card")
                print(f"Credit Card Number: {card_number}")
                print(f"Credit Card Length: {len(card_number_str)}")
                print("Credit Card Validity Status: invalid")
except ValueError:
        print("Invalid input. Please enter a valid integer.")

File: CreditCard.java
import java.util.*;

public class CreditCard{
    public static void main(String... args){

        Scanner input = new Scanner(System.in);

        System.out.print("Kindly enter your card details to confirm: ");
        String cardNumber = input.nextLine();

        if (cardNumber.length() < 13 || cardNumber.length() > 16){
            System.out.println("invalid credit card number!");
        } else{
            String cardType;
            if (cardNumber.startsWith("4")){
                cardType = "visa";
            }else if (cardNumber.startsWith("5")){
                cardType = "MasterCard";
            } else if (cardNumber.startsWith("37")){
                cardType = "American Express";
            } else if (cardNumber.startsWith("6")){
                cardType = "Discover";
            }else { 
                cardType = "unknown";
            }
int oddSum = 0;
int evenSum = 0;

for ( int i = cardNumber.length() -1; i >= 0; i -=2){
 oddSum += Character.getNumericValue(cardNumber.charAt(i));
} 

for ( int i = cardNumber.length() -2; i >= 0; i-=2){
int digit = 2 * Character.getNumericValue(cardNumber.charAt(i));
 evenSum +=  (digit / 10) + (digit % 10);
}

int totalSum = oddSum + evenSum;
if ( totalSum % 10 == 0){
        
            System.out.println("Credit Card Type: " + cardType);
	    System.out.println("Credit Card Number: " + cardNumber);
	    System.out.println("Credit Card Length: " + cardNumber.length());
	    System.out.println("Credit Card Validity Status: valid");
	  
} else {
	    System.out.println("Credit Card Type: Invalid card");
	    System.out.println("Credit Card Number: " + cardNumber);
	    System.out.println("Credit Card Length: " + cardNumber.length());
	    System.out.println("Credit Card Validity Status: invalid");


}
}
    }
}
